---
title: "Signac_analysis_LRRK2_differential_accessibility"
author: "Rebekah Langston"
date: "11/20/2020"
output: html_document
---

# Load necessary packages
```{r, message=FALSE, warning=FALSE}
library(Signac)  # v0.2.5
library(Seurat)
library(GenomeInfoDb)
library(EnsDb.Hsapiens.v86)

library(tidyverse)
library(ggpubr)
library(svglite)
library(patchwork)
set.seed(1234)
```

# Load named, clustered object
```{r}
load(file = "seurat_objects/FC_atac_named.Rdata")
FC.atac
```

# Visualize distribution of LRRK2 expression
```{r}
DefaultAssay(FC.atac) <- 'RNA'

FeaturePlot(FC.atac, "LRRK2", cols = c("#CCFFFF", "lightgrey", "#FF0000"), order = T, label = TRUE)
```

```{r}
VlnPlot(FC.atac, "LRRK2", pt.size = 0) + NoLegend()
```

# Find differentially accessible peaks between MGL.4 and all other clusters
```{r, message=FALSE, warning=FALSE}
# switch back to working with peaks instead of gene activities
DefaultAssay(FC.atac) <- 'peaks'
```

```{r}
da_peaks <- FindMarkers(
  object = FC.atac,
  ident.1 = "MGL.4",
  min.pct = 0.2,
  test.use = 'LR',
  latent.vars = 'peak_region_fragments'
)

head(da_peaks)[ , c(2, 5)]
```

# Find the closest gene to each of these peaks using the ClosestFeature function and providing an EnsDb annotation
```{r}
gene.ranges <- genes(EnsDb.Hsapiens.v86)
seqlevelsStyle(gene.ranges) <- 'UCSC'

open_MGL.4 <- rownames(da_peaks[da_peaks$avg_logFC > 0.5, ])

closest_genes_MGL.4 <- ClosestFeature(
  regions = open_MGL.4, 
  annotation = gene.ranges,
  sep = c(':', '-')
)

head(closest_genes_MGL.4, 40)[ ,c(5, 7:8)]
```

# Create coverage plots grouped by cluster around any genomic region using the CoveragePlot function
## All cells within a cluster have been averaged together, in order to visualize chromatin landscape
```{r}
# set plotting order
levels(FC.atac) <- c("MGL.4", "AST.3", "OPC.6", "ODC.0", "ODC.2", "ODC.5", "EC.13", "ExN.1", "ExN.7", "ExN.12", "InN.8", "InN.9", "InN.10", "InN.11")

# LRRK2 promoter region
CoveragePlot(
  object = FC.atac,
  region = "chr12-40206700-40228700",
  peaks = StringToGRanges(rownames(FC.atac), sep = c(":", "-")),
  annotation = gene.ranges
)
```

# Print da peaks in MGL.4 that are near LRRK2
```{r}
da_peaks_chr12 <- da_peaks[grepl( "chr12:402", rownames(da_peaks) ), ]
da_peaks_chr12[ , c(2, 5)]
```

# Examine relationship between rs76904798 genotype and LRRK2 accessibility in a population
```{r}
# Pull average expression of LRRK2 per donor in each cluster
CCvsTT_LRRK2_av.expression <- AverageExpression(FC.atac, "RNA", "LRRK2", add.ident = "dataset", return.seurat = FALSE) %>% base::as.data.frame()
```

```{r}
CCvsTT_LRRK2 <- as_tibble(CCvsTT_LRRK2_av.expression)
```

# Pull out cluster info for each donor
```{r}
S5079_TT <- CCvsTT_LRRK2 %>% select(contains("S5079"))
S1230_TT <- CCvsTT_LRRK2 %>% select(contains("S1230"))

S794_CC <- CCvsTT_LRRK2 %>% select(contains("S794"))
S1274_CC <- CCvsTT_LRRK2 %>% select(contains("S1274"))
```

# Rename columns to be cluster ID only, add donor and genotype as columns
```{r}
names(S5079_TT) = gsub(pattern = "RNA.", replacement = "", x = names(S5079_TT))
names(S5079_TT) = gsub(pattern = "_.*", replacement = "", x = names(S5079_TT))
S5079_TT <- add_column(S5079_TT, Donor = "S5079", Genotype = "TT", .before = TRUE)

names(S1230_TT) = gsub(pattern = "RNA.", replacement = "", x = names(S1230_TT))
names(S1230_TT) = gsub(pattern = "_.*", replacement = "", x = names(S1230_TT))
S1230_TT <- add_column(S1230_TT, Donor = "S1230", Genotype = "TT", .before = TRUE)


names(S794_CC) = gsub(pattern = "RNA.", replacement = "", x = names(S794_CC))
names(S794_CC) = gsub(pattern = "_.*", replacement = "", x = names(S794_CC))
S794_CC <- add_column(S794_CC, Donor = "S794", Genotype = "CC", .before = TRUE)

names(S1274_CC) = gsub(pattern = "RNA.", replacement = "", x = names(S1274_CC))
names(S1274_CC) = gsub(pattern = "_.*", replacement = "", x = names(S1274_CC))
S1274_CC <- add_column(S1274_CC, Donor = "S1274", Genotype = "CC", .before = TRUE)
```

# Join
```{r echo=T, results = "hide", message = FALSE, warning = FALSE}
CCvsTT_LRRK2_sorted <- full_join(S5079_TT, S1230_TT)
CCvsTT_LRRK2_sorted <- full_join(CCvsTT_LRRK2_sorted, S794_CC)
CCvsTT_LRRK2_sorted <- full_join(CCvsTT_LRRK2_sorted, S1274_CC)
```

# Make boxplot for microglia population MGL.4, in which we observe rs76904798:LRRK2 QTL
```{r}
MGL.4 <- CCvsTT_LRRK2_sorted %>% select(Donor, Genotype, MGL.4)

ggplot(MGL.4, aes(Genotype, MGL.4)) +
  geom_boxplot() +
  geom_jitter(shape = 21, color = "black", fill = "gray", size = 3, stroke = 1, width = 0.1) +
  labs(title = "LRRK2 expression per Donor in MGL.4", y = "Average LRRK2 Expression") +
  theme_classic()
```

# Linear regression model
```{r}
MGL.4 %>% mutate(Num.min.alleles = case_when(
  Genotype == "CC" ~ 0,
  Genotype == "TT" ~ 2)) %>%
  ggscatter(x = "Num.min.alleles", y = "MGL.4",
          color = "black", shape = 21, size = 3, add = "reg.line", 
          add.params = list(color = "blue", fill = "lightgray"),
          conf.int = TRUE, cor.coef = TRUE, cor.coeff.args = list(method = "pearson", label.x = 1.5, label.sep = "\n")) +
  theme_classic() +
  labs(x = "Number of Minor Alleles", y = "Average LRRK2 Expression")
```
```{r}
MGL.4num <- MGL.4 %>% mutate(Num.min.alleles = case_when(
  Genotype == "CC" ~ 0,
  Genotype == "CT" ~ 1,
  Genotype == "TT" ~ 2))

ct.MGL4num <- cor.test(MGL.4num$Num.min.alleles, MGL.4num$MGL.4, method = "pearson")
print(ct.MGL4num, digits = 10)
```

# Unpaired t-test with Welch's correction
```{r}
CC.MGL4 <- filter(MGL.4, Genotype == "CC")
TT.MGL4 <- filter(MGL.4, Genotype == "TT")

t.test(CC.MGL4$MGL.4, TT.MGL4$MGL.4)
```



